tls/s2n_resume.c

```diff
@@ -142,3 +144,155 @@ int s2n_store_to_cache(struct s2n_connection *conn)
 
     return 0;
 }
+
+/* This function is used in s2n_encrypt_session_ticket in order for s2n to
+ * choose a valid encryption key from all of the keys added to config
+ */
+struct s2n_ticket_key *s2n_get_valid_ticket_key(struct s2n_config *config)
+{
+    /* Currently, the first key is returned and assumed to be valid.
+     * In the future, this function may choose between multiple valid keys,
+     * which would allow s2n to phase keys in and out linearly.
+     *
+     * if (config->num_prepped_ticket_keys > 0) {
+     *     return &config->ticket_keys[0];
+     * }
+     */
+
+    /* No valid keys added */
+    return NULL;
+}
+
+/* This function is used in s2n_decrypt_session_ticket in order for s2n to
+ * find the matching key that was used for encryption.
+ */
+struct s2n_ticket_key *s2n_find_ticket_key(struct s2n_config *config, uint8_t name[16])
+{
+    /* for (int i = 0; i < config->num_prepped_ticket_keys; i++) {
+     *     if (memcmp(config->ticket_keys[i].key_name, name, 16) == 0) {
+     *         return &config->ticket_keys[i];
+     *     }
+     * }
+     */
+
+    /* Could not find key with that name */
+    return NULL;
+}
+
+int s2n_encrypt_session_ticket(struct s2n_connection *conn, struct s2n_stuffer *to)
```

Comment:
Would it be a better idea to make this take a ticket key as argument? We could have the possibly complex logic used to select a ticket key outside.